python_modules/dagster/dagster/_utils/cached_method.py

```diff
@@ -43,18 +51,18 @@ def a_method(self, arg1, arg2):
     """
 
     @wraps(method)
-    def helper(self, *args, **kwargs):
+    def helper(self: S, *args: P.args, **kwargs: P.kwargs) -> T:
         cache_attr_name = method.__name__ + "_cache"
         if not hasattr(self, cache_attr_name):
-            cache = {}
+            cache: Dict[Hashable, T] = {}
             setattr(self, cache_attr_name, cache)
         else:
             cache = getattr(self, cache_attr_name)
 
         key = _make_key(args, kwargs)
         cached_result = cache.get(key, NO_VALUE_IN_CACHE_SENTINEL)
         if cached_result is not NO_VALUE_IN_CACHE_SENTINEL:
```

Comment:
bit confused why this just isn't:

```python
if key in cache:
   return cache[key]
```

I guess one could argument perf but having the sentinel object seems unnecessarily noisy

Reply:
that is exactly the change I made upstack in the PR that made the runtime change